pyposeidon/dem.py

```diff
@@ -45,14 +45,30 @@ def __init__(self, dem_source=None, **kwargs):
         else:
             self.Dataset = dem_(source=dem_source, **kwargs)
 
-        coastline = kwargs.get("coastlines", None)
-        if coastline is not None:
-            if kwargs.get("adjust_dem", True):
+        #        print('CHECK2  :','adjusted' in self.Dataset.data_vars.keys())
+        #        print(self.Dataset.data_vars.keys())
+        if kwargs.get("adjust_dem", True):
+            coastline = kwargs.get("coastlines", None)
+            if coastline is None:
+                logger.warning("coastlines not present, aborting adjusting dem\n")
+            elif "adjusted" in self.Dataset.data_vars.keys():
+                logger.info("Dem already adjusted\n")
+            elif "fval" in self.Dataset.data_vars.keys():
+                logger.info("Dem already adjusted\n")
+            else:
```

Comment:
Minor nitpick 1: I would add a `log.info("Adjusting DEM")` or similar in the `else` block